### PR TITLE
Filter

### LINKED ISSUE #5: [US-06] Filter by category
## Value Statement (User Story)

As a user I want to be able to filter the products by category.

## Description

filter-function

## Acceptance Criteria

- [x] A filter button should appear on top of the page
- [x] The filter button should open a dropdown menu with different categories
- [x] When clicking on one of the categories, the product list should show products from that category

## Tasks

- [x] Create feature branch
- [x] Build FilterButton component
- [x] Add FilterButton to product list page
- [x] Create dropdown menu with categories
- [x] Write filter function
- [x] Add filter function on click to the categories
- [x] Write test for the new component

## Complexity
Medium

### PR DESCRIPTION
**Pull request by mortaen**
Added a button on the products page which opens a menu with categories to filter the products. Also, when adding a new product via the form, the submitted categories will now appear in the filter menu and be stored locally.
Resolves #5

### CODE REVIEW

File: src/components/Filter/Filter.spec.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      />
+    )
+
+    const button = screen.getByLabelText('show filters')
+    expect(button).toBeInTheDocument()
+    userEvent.click(button)
+
+    const listItems = screen.getAllByRole('listitem')
+    expect(listItems).toHaveLength(4)
+  })
```
Comment (timurbuch) [inline src/components/Filter/Filter.spec.js]: That's a nice test 👍🏾

File: src/components/Navigation/Navigation.js [hunk 2]
```diff
@@ -57,6 +57,11 @@ const NavItem = styled.li`
   padding: 5px;
 `
 
+const StyledLink = styled(Link)`
+  text-decoration: none;
+  color: inherit;
+`
+
```
Comment (timurbuch) [inline src/components/Navigation/Navigation.js:64]: Nice 👍🏾

File: src/App.js [hunk 3]
```diff
@@ -15,6 +15,16 @@ function App({ initialData }) {
...
+  const initialCategories = ['plant milk', 'plant yoghurt', 'meat alternative']
+
+  const [filterCategories, setFilterCategories] = useState(() => {
+    if (localStorage.getItem('categoriesLocalStorage')) {
+      return JSON.parse(localStorage.getItem('categoriesLocalStorage'))
+    } else {
+      return initialCategories
+    }
+  })
+
```
Comment (timurbuch) [inline src/App.js:27]: Do you need all of this in the App.js? Might be better to put it in its file, to not loose overview.

**Review by timurbuch — COMMENTED** [review]
Looks pretty good, I added some remarks 👍🏾

File: src/App.js (hunk 3 above)
Comment (mortaen) [reply src/App.js:27]: Will probably end up creating folders for the hooks and outsourcing them that way, I agree with your assessment, it's getting a bit crowded in the App.js 😸

**Pull request merged**